app/src/main/java/com/simplecity/amp_library/utils/DataManager.java

```diff
@@ -0,0 +1,359 @@
+package com.simplecity.amp_library.utils;
+
+import android.annotation.SuppressLint;
+import android.database.ContentObserver;
+import android.net.Uri;
+import android.os.Handler;
+import android.os.Looper;
+import android.provider.MediaStore;
+
+import com.annimon.stream.Optional;
+import com.annimon.stream.Stream;
+import com.annimon.stream.function.Predicate;
+import com.jakewharton.rxrelay2.BehaviorRelay;
+import com.simplecity.amp_library.ShuttleApplication;
+import com.simplecity.amp_library.model.Album;
+import com.simplecity.amp_library.model.AlbumArtist;
+import com.simplecity.amp_library.model.Genre;
+import com.simplecity.amp_library.model.InclExclItem;
+import com.simplecity.amp_library.model.Playlist;
+import com.simplecity.amp_library.model.Song;
+import com.simplecity.amp_library.sql.databases.BlacklistWhitelistDbOpenHelper;
+import com.simplecity.amp_library.sql.sqlbrite.SqlBriteUtils;
+import com.squareup.sqlbrite2.BriteDatabase;
+import com.squareup.sqlbrite2.SqlBrite;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableTransformer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.annotations.Nullable;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+
```

Comment:
Could you add comment block for this class explaining what it is and what it does? We also need a license header for this class and a comment for where the code came from, assuming you didn't write it.